src/core_plugins/kibana/public/dashboard/panel/panel_header/panel_actions/get_inspector_panel_action.tsx

```diff
@@ -47,37 +47,35 @@ export function getInspectorPanelAction({
     {
       icon: <EuiIcon type="inspect" />,
       isVisible: ({ embeddable }) => {
-        if (!embeddable) {
-          return false;
+        if (embeddable && embeddable.handler) {
+          return Inspector.isAvailable(embeddable.handler.inspectorAdapters);
         }
-        return Inspector.isAvailable(embeddable.getInspectorAdapters());
+        return false;
       },
       onClick: ({ embeddable }) => {
-        if (!embeddable) {
-          return;
-        }
-        closeContextMenu();
-        const adapters = embeddable.getInspectorAdapters();
-        if (!adapters) {
-          return;
-        }
-
-        const session = Inspector.open(adapters, {
-          title: panelTitle,
-        });
-        // Overwrite the embeddables.destroy() function to close the inspector
-        // before calling the original destroy method
-        const originalDestroy = embeddable.destroy;
-        embeddable.destroy = () => {
-          session.close();
-          if (originalDestroy) {
-            originalDestroy.call(embeddable);
+        if (embeddable && embeddable.handler) {
+          closeContextMenu();
+          const adapters = embeddable.handler.inspectorAdapters;
```

Comment:
This PR changes the embeddable interface from using `getInspectorAdapters` to just `embeddable.handler.inspectorAdapters` but only updates visualization embeddables. Saved search embeddables also need to be updated - https://github.com/elastic/kibana/blob/master/src/core_plugins/kibana/public/discover/embeddable/search_embeddable.js#L47